gpii/node_modules/flowManager/src/PSPChannel.js

```diff
@@ -76,7 +79,15 @@ fluid.defaults("gpii.pspChannel.sessionBinder", {
 fluid.defaults("gpii.pspChannel", {
     gradeNames: ["fluid.modelComponent"],
     members: {
-        outputBlocked: null
+        outputBlocked: null,
+        // "defaultSettingControls" holds schemas containing default preferences and their values
+        // defined in the reset to default file.
+        defaultSettingControls: {
+            expander: {
+                funcName: "gpii.pspChannel.generateDefaultSettingControls",
+                args: ["{flowManager}.options.defaultSettings", "{ontologyHandler}.ontologyMetadata.flat"]
```

Comment:
I'm sorry I didn't spot that this implementation now contains two important race conditions.
Firstly, this component is constructed during global startup, and the ontologyHandler only loads its metadata asynchronously during its onCreate - and so during the initial fetch of these settings the ontologyMetadata is empty.

Even worse is the issue which is the cause of https://issues.gpii.net/browse/GPII-4005 which is that we now have a global circular component startup race, through the following route:

i) We start up the server, which has as its top-level member, 
matchmaker: {
    type: "fluid.flatMatchMaker"

ii) This requests the fetch of "{flowManager}.ontologyHandler" as a result of its subcomponent definition - 
ontologyHandler: "{flowManager}.ontologyHandler"

This then synchronously schedules the construction of the flowManager, since otherwise we can't resolve the reference
iii) As part of the "onCreate" of the flowManager, we then execute "onCreate.convertDefaultSettingsToSnapshot" which then loads the default settings [note that these definitions are also race-prone since there is no ordering established between the listeners:
        "onCreate.loadDefaultSettings": {
            funcName: "fluid.set",
            args: ["{that}", ["options", "defaultSettings"], "@expand:{defaultSettingsLoader}.get()"]
        },
        // Convert the default settings to lifecycle instruction snapshot only once on system start.
        "onCreate.convertDefaultSettingsToSnapshot": {
            funcName: "gpii.flowManager.convertDefaultSettingsToSnapshot",
            // Passing "{that}.options.defaultSettings" explicitly to ensure it has been resolved by the IoC framework
            // when reaching the function
            args: ["{that}", "{that}.options.defaultSettings", "{that}.events.defaultSnapshotCreated"]
        },
]

iv) convertDefaultSettingsToSnapshot then in its impl attempts to invoke a method on the privateMatchmaker:

gpii.flowManager.convertDefaultSettingsToSnapshot = function (that, defaultSettings, defaultSnapshotCreatedEvent) {
    if (defaultSettings) {
        var matchMakingPromise = that.privateMatchMaker.doMatch("noUser", defaultSettings);

this then invokes matchMaker.match for the FlatMatchMaker, which makes a request for the {ontologyHandler} yet again in its argument list - which doesn't resolve, because we were already in the middle of instantiating the ontologyHandler as a result of the reference at point ii) !
Really the framework should report a better error at this point, but it has a faulty branch here: https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/FluidIoC.js#L903 which assumes that because there is SOMETHING mounted at the subcomponent path it must actually be a component, but in fact it is the "in evaluation marker" and it then just bombs out and returns undefined. It's not worth fixing this at this point since the "new framework" gets rid of races of this kind.

In the meantime we need to rewrite all of this structure so that it does not try to query the ontologyHandler until it is ready, which will cut both of the loops at the same time.